src/main/java/github/scarsz/discordsrv/listeners/DiscordChatListener.java

```diff
@@ -59,9 +62,16 @@
     @Override
     public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
         // if message is from null author or self do not process
-        if (event.getMember() == null || DiscordUtil.getJda() == null || event.getAuthor().equals(DiscordUtil.getJda().getSelfUser()))
+        if ((event.getMember() == null && !event.isWebhookMessage()) || DiscordUtil.getJda() == null || event.getAuthor().equals(DiscordUtil.getJda().getSelfUser()))
             return;
 
+        Webhook webhook = event.getChannel().retrieveWebhooks().complete().stream()
+                .filter(hook -> hook.getName().equals("DiscordSRV " + event.getChannel().getId()))
+                .findFirst().orElse(null);
+
+        if (webhook != null)
+            if (webhook.getId().equals(event.getAuthor().getId())) return;
```

Comment:
Use `if (condition1 && condition2)` rather than two seperate `if`s in places like this. This check is not necessary if `DiscordChatChannelBlockWebhooks` is enabled (effectively eliminating the need to know the webhook id)